src/libraries/System.Private.Uri/src/System/Uri.cs

```diff
@@ -1518,24 +1530,17 @@ public override int GetHashCode()
                 return CalculateCaseInsensitiveHashCode(OriginalString);
             }
 
-            // Consider moving hash code storage from m_Info.MoreInfo to m_Info
-            UriInfo info = EnsureUriInfo();
-            if ((object?)info.MoreInfo == null)
-            {
-                info.MoreInfo = new MoreInfo();
-            }
-            int tempHash = info.MoreInfo.Hash;
+            MoreInfo info = EnsureUriInfo().MoreInfo;
+            int tempHash = info.Hash;
             if (tempHash == 0)
             {
-                string? chkString = info.MoreInfo.RemoteUrl;
-                if ((object?)chkString == null)
-                    chkString = GetParts(UriComponents.HttpRequestUrl, UriFormat.SafeUnescaped);
+                string chkString = info.RemoteUrl ??= GetParts(UriComponents.HttpRequestUrl, UriFormat.SafeUnescaped);
                 tempHash = CalculateCaseInsensitiveHashCode(chkString);
                 if (tempHash == 0)
                 {
                     tempHash = 0x1000000;   //making it not zero still large enough to be mapped to zero by a hashtable
```

Comment:
I don't understand this comment.  Do you?

Reply:
I do not. It's been removed now that #32713 is merged.